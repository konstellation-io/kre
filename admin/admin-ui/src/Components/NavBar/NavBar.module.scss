@import 'Styles/colors';
@import 'Styles/mixins';
@import 'Styles/variables';

.disabled {
  cursor: default;
}

.item {
  @include font-button;
  @include list-item-hover;

  text-align: center;
  color: font-color(lowlight);
  line-height: 10px;

  position: relative;
  white-space: nowrap;
  cursor: pointer;

  display: flex;
  flex-direction: row;
  align-items: center;

  span {
    margin: 0 15px;
  }

  .warning {
    color: $color-warn;
  }

  &.disabled {
    pointer-events: none;
    color: font-color(dark);
  }

  &.size_s {
    height: 32px;
    padding: 0 16px;
  }

  &.size_m {
    height: 56px;
    padding: 0 28px;
  }

  &:hover {
    .arrow {
      opacity: 1;
      margin-left: auto;
    }
  }

  .arrow {
    opacity: 0;
    margin-left: auto;
    transition: opacity ease 0.2s;
  }
}

.active {
  cursor: default;
  pointer-events: none;
}

.active .item {
  background-color: $bg-highlighted;
  color: white;

  .arrow {
    opacity: 1;
    margin-left: auto;
  }
}
