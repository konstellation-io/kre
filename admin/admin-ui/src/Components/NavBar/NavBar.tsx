import { has } from 'lodash';

import React, { MouseEvent, FunctionComponent } from 'react';
import IconWarning from '@material-ui/icons/Warning';
import IconArrowForward from '@material-ui/icons/ArrowForward';
import { NavLink } from 'react-router-dom';

import styles from './NavBar.module.scss';
import cx from 'classnames';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

export enum ItemSize {
  SMALL = 'size_s',
  MEDIUM = 'size_m'
}

export type Tab = {
  label: string;
  route: string;
  Icon?: FunctionComponent<SvgIconProps>;
  exact?: boolean;
  disabled?: boolean;
  showWarning?: boolean;
  warningTitle?: string;
};
type Props = {
  tabs: Tab[];
  itemSize?: ItemSize;
  showItemArrows?: boolean;
};

function NavBar({
  tabs,
  itemSize = ItemSize.MEDIUM,
  showItemArrows = false
}: Props) {
  function handleClick(
    event: MouseEvent<HTMLAnchorElement>,
    disabled: boolean = false
  ) {
    if (disabled) {
      event.preventDefault();
    }
  }

  const tabElements = tabs.map(tab => {
    const exact = has(tab, 'exact') ? tab.exact : true;
    return (
      <NavLink
        key={`${tab.label}-${tab.route}`}
        to={tab.route}
        activeClassName={styles.active}
        exact={exact}
        onClick={(e: MouseEvent<HTMLAnchorElement>) =>
          handleClick(e, tab.disabled)
        }
        className={cx({ [styles.disabled]: tab.disabled })}
        replace
      >
        <div
          className={cx(
            styles.item,
            { [styles.disabled]: tab.disabled },
            styles[itemSize]
          )}
        >
          {tab.Icon && <tab.Icon className="icon-regular" />}
          <span>{tab.label}</span>
          {showItemArrows && (
            <div className={styles.arrow}>
              <IconArrowForward className="icon-regular" />
            </div>
          )}
          {tab.showWarning && (
            <div title={tab.warningTitle}>
              <IconWarning className={cx('icon-regular', styles.warning)} />
            </div>
          )}
        </div>
      </NavLink>
    );
  });

  return <div>{tabElements}</div>;
}

export default NavBar;
