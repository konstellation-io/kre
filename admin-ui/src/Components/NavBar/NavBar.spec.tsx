import React from 'react';
import NavBar, { Tab } from './NavBar';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';

const tabs = [
  { label: 'TAB 1', route: '/home' } as Tab,
  { label: 'TAB 2', route: '/home' } as Tab,
  { label: 'TAB 3', route: '/home' } as Tab
];

describe('NavBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar tabs={tabs} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.exists(NavLink)).toBeTruthy();
    expect(wrapper.find(NavLink).length).toBe(3);
  });

  it('Shows different options', () => {
    expect(
      wrapper
        .find(NavLink)
        .at(0)
        .find('span')
        .text()
    ).toBe('TAB 1');
    expect(
      wrapper
        .find(NavLink)
        .at(1)
        .find('span')
        .text()
    ).toBe('TAB 2');
    expect(
      wrapper
        .find(NavLink)
        .at(2)
        .find('span')
        .text()
    ).toBe('TAB 3');
  });
});
