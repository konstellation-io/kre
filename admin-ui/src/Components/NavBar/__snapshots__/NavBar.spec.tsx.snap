// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<div>
  <NavLink
    activeClassName="active"
    className=""
    exact={true}
    key="TAB 1-/home"
    onClick={[Function]}
    replace={true}
    to="/home"
  >
    <div
      className="item size_m"
    >
      <span>
        TAB 1
      </span>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    className=""
    exact={true}
    key="TAB 2-/home"
    onClick={[Function]}
    replace={true}
    to="/home"
  >
    <div
      className="item size_m"
    >
      <span>
        TAB 2
      </span>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    className=""
    exact={true}
    key="TAB 3-/home"
    onClick={[Function]}
    replace={true}
    to="/home"
  >
    <div
      className="item size_m"
    >
      <span>
        TAB 3
      </span>
    </div>
  </NavLink>
</div>
`;
